pokedex_async_redux/lib/features/pokemon_details/widgets/stats_card.dart

```diff
@@ -0,0 +1,120 @@
+import 'package:flutter/material.dart';
+import 'package:pokedex_asyn_redux/api/model/stats.dart';
+import 'package:pokedex_asyn_redux/utils/color_constants.dart';
+import 'package:pokedex_asyn_redux/utils/string_constants.dart';
+import 'package:pokedex_asyn_redux/widgets/spacing.dart';
+
+class StatsContainer extends StatelessWidget {
+  const StatsContainer({
+    Key? key,
+    required this.stat,
+  }) : super(key: key);
+
+  final List<Stats> stat;
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      children: [
+        const Text(
+          statsLabel,
+          style: TextStyle(
+            fontSize: 20.0,
+            fontWeight: FontWeight.bold,
+          ),
+        ),
+        const VerticalSpace(height: 10.0),
+        Column(
+          children: stat.map((stats) => _StatItem(stats: stats)).toList(),
+        ),
+      ],
+    );
+  }
+}
+
+class _StatItem extends StatelessWidget {
+  const _StatItem({Key? key, required this.stats}) : super(key: key);
+
+  final Stats stats;
+
+  @override
+  Widget build(BuildContext context) {
+    String imageName;
+    String title;
+    switch (stats.stat.name) {
```

Comment:
would be better if you move this in another separate private function that returns `Map<String, String>` to make this more readable 🙂 

after that then just declare it immediately in a private variable before the `return` statement like this: 

`final something = _getSomething(stats.stat.name);`